<!--
  Unified Planning Pipeline
  统一规划管线：根据 mission 参数选择后端（line/skidpad/high_speed）
-->
<launch>
  <arg name="mission" default="high_speed" doc="任务类型: line, acceleration, skidpad, high_speed, trackdrive, autocross"/>
  <arg name="output_pathlimits_topic" default="planning/pathlimits"/>
  <arg name="input_car_state_topic" default="localization/car_state"/>
  <arg name="output_approaching_goal_topic" default="planning/skidpad/approaching_goal"/>
  <arg name="diagnostics_topic" default="planning/diagnostics"/>
  <arg name="publish_global_diagnostics" default="true"/>
  <arg name="global_diagnostics_topic" default="/diagnostics"/>
  <arg name="diagnostics_rate_hz" default="1.0"/>
  <arg name="enable_internal_viz_side_channel" default="false"/>

  <node pkg="planning_ros" type="planning_pipeline_node" name="planning_pipeline" output="screen">
    <param name="mission" value="$(arg mission)"/>
    <param name="output_pathlimits_topic" value="$(arg output_pathlimits_topic)"/>
    <param name="input_car_state_topic" value="$(arg input_car_state_topic)"/>
    <param name="output_approaching_goal_topic" value="$(arg output_approaching_goal_topic)"/>
    <param name="input_pose_topic" value="$(arg input_car_state_topic)"/>
    <param name="topics/car_state" value="$(arg input_car_state_topic)"/>
    <param name="topics/pathlimits" value="$(arg output_pathlimits_topic)"/>
    <param name="topics/approaching_goal" value="$(arg output_approaching_goal_topic)"/>
    <param name="diagnostics_topic" value="$(arg diagnostics_topic)"/>
    <param name="publish_global_diagnostics" value="$(arg publish_global_diagnostics)"/>
    <param name="global_diagnostics_topic" value="$(arg global_diagnostics_topic)"/>
    <param name="diagnostics_rate_hz" value="$(arg diagnostics_rate_hz)"/>
    <param name="enable_internal_viz_side_channel" value="$(arg enable_internal_viz_side_channel)"/>

    <!-- 按 mission 加载对应 config 到节点私有命名空间 -->
    <rosparam if="$(eval arg('mission') in ('line', 'acceleration'))"
              command="load" file="$(find planning_ros)/config/line_detection.yaml"/>
    <rosparam if="$(eval arg('mission') == 'skidpad')"
              command="load" file="$(find planning_ros)/config/skidpad_detection.yaml"/>
    <rosparam if="$(eval arg('mission') in ('high_speed', 'trackdrive', 'autocross'))"
              command="load" file="$(find planning_ros)/config/high_speed_tracking.yaml"/>
  </node>
</launch>
